### ISSUE TITLE
Complete dataset and code of dataset generation for the Scene Completion

### ISSUE THREAD

**Issue by SugTak0806** [PRBonn/semantic-kitti-api#33]
Hi. Thank you for sharing the great job!

1. Where is the full training dataset of the Semantic Scene Completion?
I got the dataset from http://www.semantic-kitti.org/assets/data_odometry_voxels.zip .
But the the dataset contains every 5th scans only. (01/voxels/000000.bin, 01/voxels/000005.bin, ...)
How can I get the all scans?

2.  The dataset generation code of the Semantic Scene Completion
Could you share the code for generating the voxel data from the raw point cloud?

Thanks.

**Comment by jbehley** [comment]
Thanks for your interest in our scene completion dataset.

Regarding your questions:
1. We currently have no plans to provide all data due to the size of the data.
2. We have to discuss if we can and want to provide the tool to generate the voxels. It might take some time to polish the code and refactor it.

**Comment by joanshen0508** [comment]
Hi, many thanks for sharing the data. 

We hope to follow your work of completion and reimplement your method, as the released data contains every 5th scan only, did you only use these data to training the completion model? Or the number of training data used in completion is the same as that used in the segmentation task?

Thanks.

**Comment by joanshen0508** [comment]
> Thanks for your interest in our scene completion dataset.
> 
> Regarding your questions:
> 
> 1. We currently have no plans to provide all data due to the size of the data.
> 2. We have to discuss if we can and want to provide the tool to generate the voxels. It might take some time to polish the code and refactor it.

Hi, many thanks for sharing the data.

We hope to follow your work of completion and reimplement your method, as the released data contains every 5th scan only, did you only use these data to training the completion model? Or the number of training data used in completion is the same as that used in the segmentation task?

Thanks. @jbehley

**Comment by jbehley** [comment]
We used exactely this data to train the networks. (It is unfortunately not very clear from the paper.)

I will have maybe next week a look in the voxelizer such that you can generate more training data.

**Comment by joanshen0508** [comment]
> We used exactely this data to train the networks. (It is unfortunately not very clear from the paper.)
> 
> I will have maybe next week a look in the voxelizer such that you can generate more training data.

Many thanks for your reply,

In the paper, it seems that 19 130 pairs of input and target voxel grids are used for training, 815 for validation and 3 992 for testing.
The number of validation and testing is right (even every 5th scan). Do you mean that you use only 3000+ (not 19130) data for training?

Thanks. @jbehley

**Comment by jbehley** [comment]
As far as I know, we used the 3000+ scans for training. As far as I know, it was anyway needed to train on parts of the data due to memory restrictions.

Hope that helps.

**Comment by agiachris** [comment]
Hi @jbehley , 

Thanks for providing this!

I just have one question that is a little unclear from the paper. It is mentioned that the voxelized spatial extents for SSC task is x = [0, 51.2], y=[-25.6, 25.6], however, for the z component, it only mentions 6.4 m discretized at 0.2 (i.e. 32 channels). 

Could you provide the point cloud spatial extents in the z component (e.g. z_min, z_max), that would correspond to the 32 height channel voxelization, such that the voxelized point cloud would align with the  .label files?

Thanks in advance.

**Comment by jbehley** [comment]
```
min extent: [0, -25.6, -2] 
max extent: [51.2, 25.6,  4.4] 
voxel size: 0.2
```

were the settings from our configuration file, i.e.,  we used -2.0 m to 4.4 m.

I still had other stuff with higher priority, but I hope that I can finish until the end of the week the cleanup of the voxelizer code. Sorry for the delay.

**Comment by agiachris** [comment]
@jbehley - much appreciated.
No worries, and all the best.

**Comment by jbehley** [comment]
I finally managed to clean up the code a bit and implement the saving as binary files. You can find the code, we used for generating the voxel grids under https://github.com/jbehley/voxelizer

**Comment by jbehley** [comment]
Since the voxelized is now available, I close this issue now. Reopen the issue if you need additional help.
